gui/dewar_tree.py

```diff
@@ -89,173 +97,181 @@ def keyPressEvent(self, event):
+                current_flags = parentItem.flags()
+                parentItem.setFlags(current_flags & ~Qt.ItemFlag.ItemIsSelectable)  # type: ignore
+                position_s = f'{j+1}-{sample.get("name", "")}'
+                item = QtGui.QStandardItem(
+                    QtGui.QIcon(ICON),
+                    position_s,
+                )
+                return
+
+            position_s = f'{j+1}-{sample.get("name", "")}'
+            item = QtGui.QStandardItem(
+                QtGui.QIcon(ICON),
+                position_s,
+            )
+            # just stuck sampleID there, but negate it to diff from reqID
+            item.setData(sample_id, 32)
+            item.setData("sample", 33)
+            if sample_id == self.parent.mountedPin_pv.get():
+                self.set_mounted_sample(item)
+            parentItem.appendRow(item)
+            if sample_id == self.parent.mountedPin_pv.get():
+                mountedIndex = self.model.indexFromItem(item)
+            # looking for the selected item
+            if sample_id == self.parent.selectedSampleID:
+                logger.info("found " + str(self.parent.SelectedItemData))
+                selectedSampleIndex = self.model.indexFromItem(item)
+            sampleRequestList = db_lib.getRequestsBySampleID(sample_id)
+            for request in sampleRequestList:
+                if not ("protocol" in request["request_obj"]):
+                    continue
+                col_item = self.create_request_item(request)
+                if request["priority"] == 99999:
+                    selectedIndex = self.model.indexFromItem(
+                        col_item
+                    )  ##attempt to leave it on the request after collection
+                    collectionRunning = True
+                item.appendRow(col_item)
+                if (
+                    request["uid"] == self.parent.SelectedItemData
+                ):  # looking for the selected item, this is a request
+                    selectedIndex = self.model.indexFromItem(col_item)
+
+        current_index = None
+        if not collectionRunning:
+            if selectedSampleIndex:
+                current_index = selectedSampleIndex
+            elif mountedIndex:
+                current_index = mountedIndex
+                item = self.model.itemFromIndex(mountedIndex)
+                self.set_mounted_sample(item)
+            elif selectedIndex:
+                current_index = selectedIndex
+        elif collectionRunning and mountedIndex:
+            current_index = mountedIndex
+
+        if current_index:
+            self.setCurrentIndex(current_index)
+            self.parent.row_clicked(current_index)
+
+    def is_proposal_member(self, proposal) -> bool:
+        # Check if the user running LSDC is part of the sample's proposal
+        if proposal not in self.proposal_membership:
+            r = requests.get(f"{os.environ['NSLS_API_URL']}/proposal/{proposal}")
```

Comment:
NSLS_API_URL should be renamed to NSLS2_API_URL and must be required (check it).
also, the "proposal" field name is too generic - this should be called proposal_id